projects/packages/stats-admin/src/class-admin-post-list-column.php

```diff
@@ -89,7 +89,30 @@ public function add_stats_post_table_cell( $column, $post_id ) {
 				);
 			} else {
 				// Link to the wp-admin stats page.
-				$stats_post_url = admin_url( sprintf( 'admin.php?page=stats#!/stats/post/%d/%d', $post_id, \Jetpack_Options::get_option( 'id', 0 ) ) );
+				$query_args = array(
+					'from'         => 'postList',
+					'jp_post_type' => get_post_type( $post_id ),
+				);
+
+				$list_criteria_params = array(
+					's'             => sanitize_text_field( get_search_query() ),
+					'paged'         => absint( get_query_var( 'paged' ) ),
+					'post_status'   => sanitize_text_field( get_query_var( 'post_status' ) ),
+					'orderby'       => sanitize_text_field( get_query_var( 'orderby' ) ),
+					'order'         => sanitize_text_field( get_query_var( 'order' ) ),
+					'author'        => absint( get_query_var( 'author' ) ),
+					'cat'           => absint( get_query_var( 'cat' ) ), // 'cat' is the query var for category ID
+					'm'             => absint( get_query_var( 'm' ) ),   // 'm' is the query var for YYYYMM
+					'category_name' => sanitize_text_field( get_query_var( 'category_name' ) ),
+				);
+
+				foreach ( $list_criteria_params as $key => $value ) {
+					if ( isset( $_GET[ $key ] ) && $value ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Checking if the key existts and not reading the value from the request.
```

Comment:
```suggestion
					if ( isset( $_GET[ $key ] ) && $value ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Checking if the key exists and not reading the value from the request.
```

Reply:
1. What do you think of serialize all these params into a single param to pass on to Stats dashboard so that they wouldn't interfere with the target page?

For example, we add a param called `postParams` with value of `url_encode(... get the prams in $_GET here ...)`

2. I would recommend to use a fixed list of params to reduce the risk of being hijacked.